### PR TITLE
Added map styling

### PR DESCRIPTION
**Pull request by azt3k**

### CODE REVIEW

**Comment by gilbox** [comment]
If I understand this correctly you are not handling differences between iOS and android? What do you think about doing that?

File: android/build.gradle [hunk 1]
```diff
@@ -59,7 +59,6 @@ android {
 }
 
 dependencies {
-  compile 'com.facebook.react:react-native:+'
-  compile "com.google.android.gms:play-services-base:9.4.0"
-  compile 'com.google.android.gms:play-services-maps:9.4.0'
+    compile 'com.facebook.react:react-native:+'
+    compile 'com.google.android.gms:play-services:9.6.1'
```
Comment (azt3k) [inline android/build.gradle]: 9.6 is needed for the map styling https://developers.google.com/android/guides/releases

Comment (azt3k) [inline android/build.gradle]: This also works for iOS, but only with the google maps provider.

File: android/src/main/java/com/airbnb/android/react/maps/AirMapManager.java [hunk 2]
```diff
@@ -20,6 +20,9 @@
 import com.google.android.gms.maps.MapsInitializer;
 import com.google.android.gms.maps.model.LatLng;
 import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gms.maps.model.MapStyleOptions;
+import android.util.Log;
```
Comment (felipecsl) [inline android/src/main/java/com/airbnb/android/react/maps/AirMapManager.java]: this doesnt seem to be used

File: android/build.gradle [hunk 3]
```diff
@@ -40,7 +40,7 @@ allprojects {
 
 android {
   compileSdkVersion 23
-  buildToolsVersion "23.0.3"
+  buildToolsVersion '23.0.3'
```
Comment (felipecsl) [inline android/build.gradle]: unnecessary change

File: android/build.gradle (hunk 1 above)
Comment (felipecsl) [inline android/build.gradle]: undo indentation changes

File: android/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -3,4 +3,5 @@
     package="com.airbnb.android.react.maps" >
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
Comment (felipecsl) [inline android/src/main/AndroidManifest.xml]: Do we need to handle the new permission request?

Comment (azt3k) [inline android/src/main/AndroidManifest.xml]: I believe this is already being handled in AirMapView:

````java
    private static final String[] PERMISSIONS = new String[] {
            "android.permission.ACCESS_FINE_LOCATION", "android.permission.ACCESS_COARSE_LOCATION"};
````

**Comment by azt3k** [comment]
Any update here, I've also added a bunch other features missing from the google maps implementation;

- marker anchoring
- animateToRegion
- minDelta and maxDelta
- showUserLocation

**Comment by gilbox** [comment]
sorry for letting this linger @azt3k ... Some of these fixes have been addressed already. At this point I'd recommend breaking this up into smaller PRs to make them easier to manage.

**Comment by gilbox** [comment]
Just merged https://github.com/airbnb/react-native-maps/pull/808 culled from this PR into master

**Comment by gilbox** [comment]
Just merged https://github.com/airbnb/react-native-maps/pull/817 culled from this PR into master.

**Comment by lelandrichardson** [comment]
I'm going to close this PR. I don't have a ton of context on this, but this looks like a lot of really great work (some of which has already been merged). If there is anything left to merge, please open a new PR rebased on master!  Thanks so much!
